MyBus/MyBus/Services/BaseNetworkService.swift

```diff
@@ -40,13 +40,66 @@ enum GeoCodingRouter:URLRequestConvertible{
     }
 }
 
-//TODO: add "case" options and implement them in URLRequest refactoring the GisService endpoints
+
 enum GISRouter:URLRequestConvertible{
     
+    static let MUNICIPALITY_BASE_URL:String = Configuration.gisMunicipalityApiURL()
+    static let MUNICIPALITY_ACCESS_TOKEN:String = Configuration.gisMunicipalityApiAccessToken()
+    static let MUNICIPALITY_WS_PATH:String = Configuration.gisMunicipalityApiWebServicePath()
+    static let MUNICIPALITY_WS_METHOD:String = "rest"
+    
+    /* streetNamesEndpointURL = "\(municipalityBaseURL)&endpoint=callejero_mgp&token=\(municipalityAccessToken)&nombre_calle=" */
+    case StreetNames(street:String)
+    
+    /* coordinateToAddressEndpointURL = "\(municipalityBaseURL)&endpoint=coordenada_calleaaltura&token=\(municipalityAccessToken)&latitud=\(latitude)&longitud=\(longitude)"*/
+    case CoordinateToAddress(latitude:Double, longitude:Double)
+    
+    /* addressToCoordinateEndpointURL = "\(municipalityBaseURL)&endpoint=callealtura_coordenada&token=\(municipalityAccessToken)" */
+    case AddressToCoordinate(address:String)
+    
     
     var URLRequest: NSMutableURLRequest {
         
-        return NSMutableURLRequest()
+        let (path, parameters, encoding, method): (String, [String: AnyObject], ParameterEncoding, Alamofire.Method) = {
+            
+            switch self {
+                case .StreetNames(let street):
+                    var params = [String:AnyObject]()
+                    params["method"] = GISRouter.MUNICIPALITY_WS_METHOD
+                    params["endpoint"] = "callejero_mgp"
+                    params["token"] = GISRouter.MUNICIPALITY_ACCESS_TOKEN
+                    params["nombre_calle"] = street
+                
+                    return (GISRouter.MUNICIPALITY_WS_PATH, params, .URL, .GET)
+                
+                case .CoordinateToAddress(let latitude, let longitude):
+                    var params = [String:AnyObject]()
+                    params["method"] = GISRouter.MUNICIPALITY_WS_METHOD
+                    params["endpoint"] = "coordenada_calleaaltura"
+                    params["token"] = GISRouter.MUNICIPALITY_ACCESS_TOKEN
+                    params["latitud"] = latitude
+                    params["longitud"] = longitude
+                    
+                    return (GISRouter.MUNICIPALITY_WS_PATH, params, .URL, .GET)
+                
+                case .AddressToCoordinate(let address):
+                    var params = [String:AnyObject]()
+                    params["method"] = GISRouter.MUNICIPALITY_WS_METHOD
+                    params["endpoint"] = "callealtura_coordenada" //callealtura o calleaaltura (?)
+                    params["token"] = GISRouter.MUNICIPALITY_ACCESS_TOKEN
+                    //address??
```

Comment:
Seba, endpoint string you used is right and for _address_ param we need to send:

- [ ] `codigocalle` param
- [ ] `altura` param

Extracted request params from API docs [here](https://s9.postimg.org/kdzo5t54f/Screenshot_2016_08_11_19_51_36.png)